src/Aspire.Cli/Backchannel/AppHostBackchannel.cs

```diff
@@ -106,6 +106,19 @@ public async Task ConnectAsync(Process process, string socketPath, CancellationT
         var stream = new NetworkStream(socket, true);
         var rpc = JsonRpc.Attach(stream, target);
 
+        var capabilities = await rpc.InvokeWithCancellationAsync<string[]>(
+            "GetCapabilitiesAsync",
+            Array.Empty<object>(),
+            cancellationToken);
+
+        if (!capabilities.Any(s => s == "baseline.v0"))
+        {
+            throw new AppHostIncompatibleException(
+                $"AppHost is incompatible with the CLI. The AppHost must be updated to a version that supports the baseline.v0 capability.",
```

Comment:
Is this the error that shows up in the CLI?

Reply:
Yep

Reply:
Actually its not. We write a custom error:

![image](https://github.com/user-attachments/assets/fb4d73e4-05f8-4fac-bae4-54496cc6be56)

Note this is an induced error these CLI and app host versions are compatible I just put a fake capability in to trigger this screenshot so you can get a sense of the output now.